app/src/main/kotlin/cz/covid19cz/erouska/db/SharedPrefsRepository.kt

```diff
@@ -393,11 +409,76 @@ class SharedPrefsRepository @Inject constructor(@ApplicationContext c: Context)
         return prefs.edit().putBoolean(EFGS_INTRODUCED, value).apply()
     }
 
-    fun wasHowItWorksShown() : Boolean{
+    fun wasHowItWorksShown(): Boolean {
         return prefs.getBoolean(HOW_IT_WORKS_SHOWN, false)
     }
 
-    fun setHowItWorksShown(){
+    fun setHowItWorksShown() {
         prefs.edit().putBoolean(HOW_IT_WORKS_SHOWN, true).apply()
     }
+
+    fun setVerificationData(code: String, token: String) {
+        prefs.edit().putString(VALIDATION_CODE, code)
+            .putString(VALIDATION_TOKEN, token).apply()
+    }
+
+    fun getVerificationCode(): String? {
+        return prefs.getString(VALIDATION_CODE, null)
+    }
+
+    fun getVerificationToken(): String? {
+        return prefs.getString(VALIDATION_TOKEN, null)
+    }
+
+    fun deletePublishKeysTemporaryData() {
+        prefs.edit().remove(VALIDATION_CODE)
+            .remove(VALIDATION_TOKEN)
+            .remove(TRAVELLER)
+            .remove(CONSENT_TO_FEDERATION)
+            .remove(SYMPTOM_DATE)
+            .apply()
+    }
+
+    fun isCodeValidated(code : String?): Boolean {
+        val savedCode = prefs.getString(VALIDATION_CODE, null)
+        return if (savedCode == code) {
+            hasValidationToken(useLeeway = true)
+        } else {
+            false
+        }
+    }
+
+    fun hasValidationToken(useLeeway : Boolean) : Boolean{
+        val token = prefs.getString(VALIDATION_TOKEN, null)
+        return if (token != null) {
+            !JWT(token).isExpired(if (useLeeway) AppConfig.validationTokenExpirationLeewayMinutes * 60 else 60)
```

Comment:
Chtělo by to sem comment proč je ten leewey potřeba a kdy nastane expirace

Reply:
ok